### PR TITLE
[MMM-13517] DRUM Support for new text generation target type

### PR DESCRIPTION
**Pull request by amarmudrankit**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Supporting new target type "Text Generation" in DRUM

## Rationale
Gen AI LLM model predictions are `str` data type.  We need changes to DRUM to be able to support those Completions

### CODE REVIEW

**Comment by amarmudrankit** [comment]
jenkins run test_integration_java_environment

**Comment by amarmudrankit** [comment]
jenkins please test this

**Comment by amarmudrankit** [comment]
jenkins run test_integration_r_environment

**Comment by amarmudrankit** [comment]
jenkins run test_integration_r_environment

**Comment by amarmudrankit** [comment]
jenkins run test_inference_model_templates

**Comment by amarmudrankit** [comment]
jenkins run test_inference_model_templates

**Comment by yakov-g** [comment]
jenkins run test_inference_model_templates

**Comment by yakov-g** [comment]
jenkins run test_inference_model_templates

**Comment by rvorobii** [comment]
@yakov-g please review this PR

File: custom_model_runner/datarobot_drum/drum/adapters/cli/shared/drum_input_file_adapter.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 from typing import Optional
 
 import pandas as pd
-from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX
+from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX, TEXT_GENERATION_PRED_COLUMN
```
Comment (yakov-g) [inline custom_model_runner/datarobot_drum/drum/adapters/cli/shared/drum_input_file_adapter.py:12]: seems like this import is not used

**Review by yakov-g — APPROVED** [review]
Just a nit with import.

Also I STRONGLY recommend  (aka require :) )
Add an example into model_templates, to be able to run it just out-of-box.

This use case seems quite hot so it is good to have it hands-on.
Please add it and post everywhere so people could try.

**Comment by amarmudrankit** [comment]
> Add an example into model_templates, to be able to run it just out-of-box.

I will create a ticket to add the example.  Can't add it right now with this review, because not all machinery is working right now

File: custom_model_runner/datarobot_drum/drum/adapters/cli/shared/drum_input_file_adapter.py (hunk 1 above)
Comment (amarmudrankit) [reply custom_model_runner/datarobot_drum/drum/adapters/cli/shared/drum_input_file_adapter.py:12]: I will wait for other comments and will fix it with them

**Review by akshoop — APPROVED** [review]
Overall LGTM from Core Modeling, CC @eric-s-s @scottp-ml  for awareness and review

**Pull request merged**